recipes/folly/all/conanfile.py

```diff
@@ -120,90 +113,130 @@ def validate(self):
 
         if self.options.get_safe("use_sse4_2") and str(self.settings.arch) in ['x86', 'x86_64']:
-            # in folly, if simd >=sse4.2, we also needs -mfma flag to avoid compiling error.
+            tc.preprocessor_definitions["FOLLY_SSE"] = "4"
+            tc.preprocessor_definitions["FOLLY_SSE_MINOR"] = "2"
             if not is_msvc(self):
-                cmake.definitions["CMAKE_C_FLAGS"] = "-mfma"
-                cmake.definitions["CMAKE_CXX_FLAGS"] = "-mfma"
+                tc.variables["CMAKE_C_FLAGS"] = "-mfma"
+                tc.variables["CMAKE_CXX_FLAGS"] = "-mfma"
             else:
-                cmake.definitions["CMAKE_C_FLAGS"] = "/arch:FMA"
-                cmake.definitions["CMAKE_CXX_FLAGS"] = "/arch:FMA"
-
-        cmake.definitions["CMAKE_POSITION_INDEPENDENT_CODE"] = self.options.get_safe("fPIC", True)
-
-        cxx_std_flag = tools.cppstd_flag(self.settings)
-        cxx_std_value = cxx_std_flag.split('=')[1] if cxx_std_flag else "c++{}".format(self._minimum_cpp_standard)
-        cmake.definitions["CXX_STD"] = cxx_std_value
-        if is_msvc:
-            cmake.definitions["MSVC_LANGUAGE_VERSION"] = cxx_std_value
-            cmake.definitions["MSVC_ENABLE_ALL_WARNINGS"] = False
-            cmake.definitions["MSVC_USE_STATIC_RUNTIME"] = "MT" in msvc_runtime_flag(self)
-        cmake.configure()
-        return cmake
+                tc.variables["CMAKE_C_FLAGS"] = "/arch:FMA"
+                tc.variables["CMAKE_CXX_FLAGS"] = "/arch:FMA"
+
+        # Folly is not respecting this from the helper https://github.com/conan-io/conan-center-index/pull/15726/files#r1097068754
+        tc.variables["CMAKE_POSITION_INDEPENDENT_CODE"] = self.options.get_safe("fPIC", True)
+        # Relocatable shared lib on Macos
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0042"] = "NEW"
+        # Honor CMAKE_REQUIRED_LIBRARIES in check_include_file_xxx
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0075"] = "NEW"
+        # Honor BUILD_SHARED_LIBS from conan_toolchain (see https://github.com/conan-io/conan/issues/11840)
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"
+
+        cxx_std_value = self._cppstd_flag_value(self.settings.get_safe("compiler.cppstd", self._min_cppstd))
+        # 2019.10.21.00 -> either MSVC_ flags or CXX_STD
+        if is_msvc(self):
+            tc.variables["MSVC_LANGUAGE_VERSION"] = cxx_std_value
+            tc.variables["MSVC_ENABLE_ALL_WARNINGS"] = False
+            tc.variables["MSVC_USE_STATIC_RUNTIME"] = "MT" in msvc_runtime_flag(self)
+        else:
+            tc.variables["CXX_STD"] = cxx_std_value
+
+        if not self.dependencies["boost"].options.header_only:
+            tc.cache_variables["BOOST_LINK_STATIC"] = not self.dependencies["boost"].options.shared
 
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0074"] = "NEW"  # Honor Boost_ROOT set by boost recipe
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.generate()
```

Comment:
We should be using `set_property` here to override the generated CMake targets so they match upstream

Reply:
I don't follow you here, set_property would go in package_info no? 

In fact, currently we have this

```
    def package_info(self):
        self.cpp_info.set_property("cmake_file_name", "folly")
        self.cpp_info.set_property("cmake_target_name", "Folly::folly")
        self.cpp_info.set_property("pkg_config_name", "libfolly")

```
And I think they match uptream targets:

```
install(TARGETS ${FOLLY_INSTALL_TARGETS}
  EXPORT folly
  RUNTIME DESTINATION bin
  LIBRARY DESTINATION ${LIB_INSTALL_DIR}
  ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
auto_install_files(folly ${FOLLY_DIR}
  ${hfiles}
)
...
install(
  FILES ${CMAKE_CURRENT_BINARY_DIR}/folly-config.cmake
  DESTINATION ${CMAKE_INSTALL_DIR}
  COMPONENT dev
)
install(
  EXPORT folly
  DESTINATION ${CMAKE_INSTALL_DIR}
  NAMESPACE Folly::
  FILE folly-targets.cmake
  COMPONENT dev
)
```
https://github.com/facebook/folly/blob/main/CMakeLists.txt#L453